include/exadg/solvers_and_preconditioners/preconditioners/preconditioner_amg.h

```diff
@@ -154,8 +223,54 @@ class PreconditionerML : public PreconditionerBase<double>
     // re-calculate matrix
     pde_operator.calculate_system_matrix(system_matrix);
 
-    // initialize Trilinos' AMG
-    amg.initialize(system_matrix, ml_data);
+    if(ml_operator_type == MLOperatorType::Laplace)
+    {
+      // Default setup; add near null space for Laplace operator.
+      amg.initialize(system_matrix, ml_data);
```

Comment:
Is `dimension` (used in the other branch below) irrelevant for Laplace? (scalar vs. vector Laplace)
Is there a reason why we do not unify the call `amg.initialize()` and use the same code for all the variants of the operator type?

Reply:
It is in general not, now this question pops up in `OperatorBase::get_constant_modes()`:
Do we use a scalar-valued Poisson in `dim`-dimensional space the default, or the related `dim`-component problem? I went for the former, since it is exactly what we did before as it replicates the default from deal.II.
My experiments within the structure module showed, that it did not really matter if the operator is more complex, and results were actually slightly better using the scalar-valued version (so what we have selected now).

Reply:
Plus the default fits exactly the pressure Poisson problem we often solve, for the momentum or diffusion operator, we are quite wrong in any way ... but one could try to use the dim-component version. But since I did not test all the examples I went with replicating the default from before :)